> This is a pragmatic change, thanks for the fix @mathomp4 but since we don't have a nice way to have a concrete test for this, maybe a comment pointing to the issue / pr?

Well this: https://github.com/numpy/numpy/issues/26623 does seem to show the issue. But even simpler, take this code as `test.F90`:
```
module test
   implicit none
   integer :: i = 0
end module test
```
now try and run `f2py` on it assuming it needs to link to library `libfoo.bar.so`:
```
$ f2py -m test_ -lfoo.bar -c test.F90
Cannot use distutils backend with Python>=3.12, using meson backend instead.
Using meson backend
Will pass --lower to f2py
See https://numpy.org/doc/stable/f2py/buildtools/meson.html
Reading fortran codes...
	Reading file 'test.F90' (format:free)
Post-processing...
	Block: test_
			Block: test
Applying post-processing hooks...
  character_backward_compatibility_hook
Post-processing (stage 2)...
	Block: test_
		Block: unknown_interface
			Block: test
Building modules...
    Building module "test_"...
		Constructing F90 module support for "test"...
		  Variables: i
    Wrote C/API module "test_" to file "./test_module.c"
    Fortran 90 wrappers are saved to "./test_-f2pywrappers2.f90"
The Meson build system
Version: 1.4.1
Source dir: /gpfsm/dnb34/tdirs/login/discover33.240116.mathomp4/tmpti7gdlfy
Build dir: /gpfsm/dnb34/tdirs/login/discover33.240116.mathomp4/tmpti7gdlfy/bbdir
Build type: native build

meson.build:35:8: ERROR: Expecting lparen got assign.
foo.bar = declare_dependency(link_args : ['-lfoo.bar'])
        ^

A full log can be found at /gpfsm/dnb34/tdirs/login/discover33.240116.mathomp4/tmpti7gdlfy/bbdir/meson-logs/meson-log.txt
Traceback (most recent call last):
  File "/usr/local/other/GEOSpyD/24.4.0-0_py3.12/2024-05-31/bin/f2py", line 10, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/local/other/GEOSpyD/24.4.0-0_py3.12/2024-05-31/lib/python3.12/site-packages/numpy/f2py/f2py2e.py", line 766, in main
    run_compile()
  File "/usr/local/other/GEOSpyD/24.4.0-0_py3.12/2024-05-31/lib/python3.12/site-packages/numpy/f2py/f2py2e.py", line 738, in run_compile
    builder.compile()
  File "/usr/local/other/GEOSpyD/24.4.0-0_py3.12/2024-05-31/lib/python3.12/site-packages/numpy/f2py/_backends/_meson.py", line 178, in compile
    self.run_meson(self.build_dir)
  File "/usr/local/other/GEOSpyD/24.4.0-0_py3.12/2024-05-31/lib/python3.12/site-packages/numpy/f2py/_backends/_meson.py", line 171, in run_meson
    self._run_subprocess_command(setup_command, build_dir)
  File "/usr/local/other/GEOSpyD/24.4.0-0_py3.12/2024-05-31/lib/python3.12/site-packages/numpy/f2py/_backends/_meson.py", line 167, in _run_subprocess_command
    subprocess.run(command, cwd=cwd, check=True)
  File "/usr/local/other/GEOSpyD/24.4.0-0_py3.12/2024-05-31/lib/python3.12/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['meson', 'setup', 'bbdir']' returned non-zero exit status 1.
```

It's not like `f2py` is failing because it can't find the library. It's failing because it's producing illegal meson.

From my reading of the [Meson Grammar](https://mesonbuild.com/Syntax.html#grammar), a dot in a variable means a method expression:
```
method_expression: postfix_expression "." function_expression
```
so:
```meson
foo.bar = ...
```
dies because meson is expecting `bar` to be a function and per the Grammar:
```
function_expression: id_expression "(" [argument_list] ")"
```
that means the next character must be a left paren.

--- a/numpy/f2py/tests/test_regression.py
+++ b/numpy/f2py/tests/test_regression.py
@@ -108,3 +108,14 @@ class TestF90Contiuation(util.F2PyTest):
         res=self.module.testsub(x1, x2)
         assert(res[0] == 8)
         assert(res[1] == 15)
+
+def test_gh26623():
+    # Including libraries with . should not generate an incorrect meson.build
+    try:
+        aa = util.build_module(
+            [util.getpath("tests", "src", "regression", "f90continuation.f90")],
+            ["-lfoo.bar"],
+            module_name="Blah",
+        )
+    except RuntimeError as rerr:
+        assert "lparen got assign" not in str(rerr)

